Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs

```diff
@@ -66,37 +67,23 @@ internal BrowserCommandResult<bool> InitializeModes()
         public string[] OnlineDomains { get; set; }
 
         #region PageWaits
-
-        internal void WaitForLoginPage()
-        {
-            IWebDriver driver = this.Browser.Driver;
-
-            driver.WaitUntilVisible(By.XPath(Elements.Xpath[Reference.Login.CrmMainPage])
-                , TimeSpan.FromSeconds(60),
-                e =>
-                {
-                    //determine if we landed on the Unified Client Main page
-                    if (driver.HasElement(By.XPath(Elements.Xpath[Reference.Login.CrmUCIMainPage])))
-                    {
-                        driver.WaitForPageToLoad();
-                        driver.WaitForTransaction();
-                    }
-                    else //else we landed on the Web Client main page or app picker page
-                        SwitchToDefaultContent(driver);
-                },
-                "Login page failed."
-            );
-        }
-
-        internal void WaitForMainPage()
-        {
-            IWebDriver driver = this.Browser.Driver;
-            driver.WaitUntilVisible(By.XPath(Elements.Xpath[Reference.Login.CrmMainPage]));
-            driver.WaitForPageToLoad();
-            if (driver.HasElement(By.XPath(Elements.Xpath[Reference.Login.CrmUCIMainPage])))
-            {
-                driver.WaitForTransaction();
-            }
+        internal bool WaitForMainPage(TimeSpan timeout, string errorMessage)

```

Comment:
This method was refactored,  to be more flexible, is used in Login logic to wait